### PR TITLE
dash/data-connector

### PR DESCRIPTION
**Pull request by bre1470**
Added DataPool.isNewConnector and fixed connector replacements in component.

### CODE REVIEW

File: ts/Data/DataPool.ts [hunk 1]
```diff
@@ -255,6 +255,22 @@ class DataPool implements DataEvent.Emitter {
     }
 
 
+    /**
+     * Tests whether the connector has never been requested.
+     *
+     * @param {string} connectorId
+     * Name of the connector.
```
Comment (karolkolodziej) [inline ts/Data/DataPool.ts:262]:
```suggestion
     * ID of the connector.
```

File: ts/Data/DataPool.ts [hunk 2]
```diff
@@ -346,6 +363,11 @@ class DataPool implements DataEvent.Emitter {
             }
         }
 
+        if (instances[options.id]) {
+            instances[options.id].stopPolling();
```
Comment (karolkolodziej) [inline ts/Data/DataPool.ts:367]: Why do we stop pooling here? If declared, is it initialized once again? 
Maybe it's worth checking if the `polling` is even enabled first?

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
Please add the PR message.

File: ts/Data/DataPool.ts (hunk 1 above)
Comment (bre1470) [reply ts/Data/DataPool.ts:262]: Do you call it now connector ID in Dashboards?

File: ts/Data/DataPool.ts (hunk 2 above)
Comment (bre1470) [reply ts/Data/DataPool.ts:367]: Yes, here the connector gets replaced, and when it gets disposed from the data pool, then there is no point in polling anymore. Also `stopPolling` covers all conditions, no need for an extra check.

File: ts/Data/DataPool.ts (hunk 1 above)
Comment (karolkolodziej) [reply ts/Data/DataPool.ts:262]: The argument is called `connectorId` so I thought the description should be the same

**Review by karolkolodziej — APPROVED** [review]

**Comment by sebastianbochan** [comment]
I’m testing the PR and just found some errors in HC.

Running the https://jsfiddle.net/BlackLabel/837zwjpo/ return HC #14.  The points are not mapped properly, its stange, because it is the form of data like during the init.

The problem is caused by missing "orientation" param. When I added it, it worked https://jsfiddle.net/BlackLabel/0Lkwyqxo/.

**Comment by bre1470** [comment]
@sebastianbochan Okay, so I guess, some merge of defaults is missing for setConnectorOptions.

**Comment by sebastianbochan** [comment]
After some debugging, It turned out that was my stupid mistake. I updated the flag: `firstRowAsNames` from false to true, so it returned an issue. It was not the `orientation` problem.

@bre1470 sorry for it.

**Pull request merged**